src/MongoDB.Driver/Linq/Translators/AggregateLanguageTranslator.cs

```diff
@@ -1276,5 +1297,203 @@ private bool TryTranslateSumResultOperator(PipelineExpression node, out BsonValu
+            private const string MapTemplateEndDelimiter = ">#";
+            #endregion
+
+            private readonly List<string> _usedQueryMaps = new List<string>();
+
+            #region Static methods
+            private static bool TryParseTemplate(string value, ref int startIndex, out string template)
+            {
+                template = null;
+                var startIndexOf = value.IndexOf(MapTemplateStartDelimiter, startIndex, StringComparison.Ordinal);
+                var endIndexOf = value.IndexOf(MapTemplateEndDelimiter, startIndex, StringComparison.Ordinal);
+                if (startIndexOf != -1 && endIndexOf != -1 && endIndexOf > startIndexOf)
+                {
+                    endIndexOf = endIndexOf + MapTemplateEndDelimiter.Length;
+                    if (endIndexOf <= value.Length)
+                    {
+                        template = value.Substring(startIndexOf, endIndexOf - startIndexOf);
+                        startIndex = endIndexOf;
+                        return true;
+                    }
+                }
+
+                startIndex = -1;
+                return false;
+            }
+
+            private static bool TryGetTemplateElement(string value, out BsonElement element)
+            {
+                element = new BsonElement();
+                var firstLetterIndex = value.IndexOf(MapTemplateStartDelimiter, StringComparison.Ordinal);
+                if (firstLetterIndex == -1)
+                {
+                    return false;
+                }
+
+                firstLetterIndex += MapTemplateStartDelimiter.Length;
+                var lastLetterIndex = value.IndexOf(MapTemplateEndDelimiter, StringComparison.Ordinal);
+                if (lastLetterIndex == -1)
+                {
+                    return false;
+                }
+
+                if (lastLetterIndex > firstLetterIndex)
+                {
+                    var elementValue = value.Substring(firstLetterIndex, lastLetterIndex - firstLetterIndex);
+                    element = BsonDocument.Parse(elementValue).GetElement(0);
+                    return true;
+                }
+
+                return false;
+            }
+
+            private static string AddTemplateDelimiterBrackets(string value)
+            {
+                return $"{MapTemplateStartDelimiter}{value}{MapTemplateEndDelimiter}";
+            }
+
+            private static string RemoveMapPrefixIfThere(string value)
+            {
+                if (!string.IsNullOrWhiteSpace(value))
+                {
+                    var dotIndex = value.IndexOf('.');
+                    if (dotIndex != -1)
+                    {
+                        return $"{value.Substring(dotIndex + 1)}";
+                    }
+                }
+                return value;
+            }
+
+            private static Dictionary<string, List<string>> CollectTemplates(string value)
```

Comment:
In result:
Key - is an out of scope input expression parameter.
Value - is a collection with all fields which are based on the key.